### PR TITLE
Remove redundant `onCreated` event

### PR DESCRIPTION
**Pull request by DonJayamanne**
Fixes https://github.com/microsoft/vscode-jupyter/issues/11694

### CODE REVIEW

**Review by alexdima — DISMISSED** [review]

File: src/notebooks/controllers/commands/serverConnectionControllerCommands.ts [hunk 1]
```diff
@@ -283,35 +283,37 @@ export class ServerConnectionControllerCommands implements IExtensionSingleActiv
...
+                            kind: QuickPickItemKind.Separator,
+                            label: kind
+                        });
+                        quickPickItems.push(pick);
+                        kindIndexes.set(kind, quickPickItems.length);
+                    } else {
+                        quickPickItems.splice(index, 0, pick);
+                        kindIndexes.set(kind, quickPickItems.length);
+                    }
+                    changeEmitter.fire(quickPickItems);
```
Comment (rebornix) [inline src/notebooks/controllers/commands/serverConnectionControllerCommands.ts:314]: Do we want to emit the changes once all `added` controllers are handled?

Comment (DonJayamanne) [reply src/notebooks/controllers/commands/serverConnectionControllerCommands.ts:314]: I think it should be ok, I don't see how it ocul dcause any issues (i.e. the behaviour is the same even in the past)

**Review by rebornix — APPROVED** [review]

File: src/notebooks/controllers/connectionTracker.ts [hunk 2]
```diff
@@ -87,11 +87,15 @@ export class ConnectionTracker implements IExtensionSyncActivationService, IConn
...
-    private async onCreatedController(controller: IVSCodeNotebookController) {
+    private async onChangeController(e: IVSCodeNotebookControllerUpdateEvent) {
         await Promise.all(
-            workspace.notebookDocuments.map(async (notebook) => {
-                await this.updateAffinity(notebook, controller.connection);
-            })
+            e.added.map((controller) =>
+                Promise.all(
+                    workspace.notebookDocuments.map(async (notebook) => {
+                        await this.updateAffinity(notebook, controller.connection);
```
Comment (rebornix) [inline src/notebooks/controllers/connectionTracker.ts:95]: Maybe we can have a batch affinity update impl.

**Review by rebornix — APPROVED** [review]

File: src/notebooks/controllers/connectionTracker.ts (hunk 2 above)
Comment (DonJayamanne) [reply src/notebooks/controllers/connectionTracker.ts:95]: Will create a debt item for that.

**Pull request merged**